### PR TITLE
For persistent backends, don't ever unload the corresponding shared library

### PR DESCRIPTION
**Pull request by deadeyegoodwin**

### CODE REVIEW

File: src/backends/backend/triton_backend_manager.h [hunk 1]
```diff
@@ -131,6 +134,10 @@ class TritonBackend {
 
   // Opaque state associated with the backend.
   void* state_;
+
+  // Should be backend shared library be unloaded (e.g. dlclose) when
```
Comment (CoderHam) [inline src/backends/backend/triton_backend_manager.h:138]: Should "the"?

File: src/core/persistent_backend_manager.cc [hunk 2]
```diff
@@ -100,6 +100,11 @@ PersistentBackendManager::InitPersistentBackend(
     RETURN_IF_ERROR(TritonBackendManager::CreateBackend(
         backend_name, backend_dir, backend_libpath, *config, &persist_backend));
     persist_backends_.push_back(persist_backend);
+
+    // Show that persistent backend shared libraries should never be
```
Comment (CoderHam) [inline src/core/persistent_backend_manager.cc:104]: I think "Show That" is not necessary

File: src/backends/backend/triton_backend_manager.cc [hunk 3]
```diff
@@ -224,6 +226,25 @@ TritonBackend::~TritonBackend()
   LOG_STATUS_ERROR(UnloadBackendLibrary(), "failed unloading backend");
 }
 
+void
+TritonBackend::SetUnloadEnabled(const bool enable)
+{
+  // For memory leak debugging do not allow shared libraries to be
```
Comment (CoderHam) [inline src/backends/backend/triton_backend_manager.cc:232]: I believe this comment should still be under UnloadBackendLibrary()

**Review by CoderHam — CHANGES_REQUESTED** [review]
A few small fixes else LGTM

File: src/backends/backend/triton_backend_manager.cc (hunk 3 above)
Comment (deadeyegoodwin) [reply src/backends/backend/triton_backend_manager.cc:232]: It refers to the immediately following code. Will update the comment to be more clear.

**Review by CoderHam — APPROVED** [review]

**Pull request merged**